src/mds/Server.cc

```diff
@@ -2531,7 +2531,30 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
     respond_to_request(mdr, mdr->more()->peer_error);
     return;
   }
-  
+
+  Session *session = mds->get_session(req);
+  if (session && !session->info.has_feature(CEPHFS_FEATURE_ALTERNATE_NAME) &&
```

Comment:
How about only when the fscrypt is enabled should we check the code in this `if(){}` scope ?

Reply:
Actually, I thought about that too, but decided to do this feature check first because we need a lock for checking the fscrypt.  I assumed that checking the session feature is faster and the performance impact would eventually be smaller as more and more clients get this feature.

Reply:
Ah, make sense.